conjure-runtime/src/client.rs

```diff
@@ -90,6 +93,38 @@ impl ClientState {
             proxy,
         })
     }
+
+    fn rewrite_for_mesh(service_config: &ServiceConfig) -> Result<Cow<ServiceConfig>, Error> {
+        let prefix = "mesh-";
+
+        let mesh_uris = service_config
+            .uris()
+            .iter()
+            .filter(|uri| uri.scheme().starts_with(prefix))
+            .count();
+
+        if mesh_uris == 0 {
+            return Ok(Cow::Borrowed(service_config));
+        }
+
+        if service_config.uris().len() != 1 {
+            return Err(
+                Error::internal_safe("exactly one URI must be present in mesh mode")
+                    .with_safe_param("uris", service_config.uris()),
+            );
+        }
+
+        let new_uri = service_config.uris()[0].as_str()[prefix.len()..]
+            .parse()
+            .unwrap();
+
+        let config = ServiceConfigBuilder::from(service_config.clone())
+            .uris(vec![new_uri])
+            .max_num_retries(0)
+            .build();
```

Comment:
The 0.2.x version doesn't support client QoS so we don't need to worry about that.